### ISSUE TITLE
jax.numpy.sum behaving funny for non-existent axis

### ISSUE THREAD

**Issue by lumip** [google/jax#888]
When invoking the `sum` operation on a non-existent axis I don't get an error (as would be the expected result), but sums over one of the existing axes. As far as I can tell, for any array `x` with `n` axes, calls to `sum` with axis `j > n` behave as if the provided axis was `j % n`. E.g., if the array `x` has 3 axes, calls to `sum` with `j` being any multiple of three all sum over the first axis (j=0,3,6,9,... all give the same result).
This is especially noteworthy for the one-dimensional array (only one axis), where calls to `sum` with any value for `axis` always sums over the whole array.
It would be preferable if either an error was raised, or summing over a non-existing axis would be a no-op.

Might be related to #278 

example code
```
import jax.numpy as np
x = np.array([ [ [ 1, 2 ], [ 3, 4 ] ], [ [ 5, 6 ], [ 7, 8] ], [ [ 9, 10 ], [ 11, 12 ] ] ])
assert(np.all(np.sum(x, axis=3) == np.sum(x, axis=0)))
assert(np.all(np.sum(x, axis=4) == np.sum(x, axis=1)))
assert(np.all(np.sum(x, axis=5) == np.sum(x, axis=2)))
assert(np.all(np.sum(x, axis=6) == np.sum(x, axis=0)))
# and so on..
```

**Comment by hawkinsp** [comment]
Actually, this is easy to fix, much easier than the equivalent issue for out-of-bounds indexing. Jax already needs to know the concrete values of axes at trace time, so we just need to be slightly more defensive about invalid axes. This is unlike indexes, for which the out-of-bounds test would have to take place at run time.
